cockatrice/src/tab_server.cpp

```diff
@@ -139,8 +139,17 @@ void RoomSelector::joinClicked()
 void RoomSelector::joinFinished(const Response &r, const CommandContainer & /*commandContainer*/, const QVariant &extraData)
 {
     switch (r.response_code()) {
-        case Response::RespOk: break;
-        case Response::RespUserLevelTooLow: QMessageBox::critical(this, tr("Error"), tr("You do not have the proper permission to join this room.")); return;
+        case Response::RespOk:
+            break;
+        case Response::RespNameNotFound:
+            QMessageBox::critical(this, tr("Error"), tr("Failed to join the room: it doesn't exists on the server."));
+            return;
+        case Response::RespContextError:
+            QMessageBox::critical(this, tr("Error"), tr("You are already inside the room."));
```

Comment:
Instead of this, could we do a bit of validation on the client side? Check if the room tab is being displayed via the supervisor, and if it's not, show the error, if it is, focus it.

We could potentially have some other handling - if the server says we're in the room but we don't have it open locally, log an error, tell the server to leave the room, then once the "left room" response is handled, join the room again.